### DISCUSSION TITLE
Better support for (Materialized) Views

### DISCUSSION THREAD

**Discussion by baderdean** [supabase/supabase#16389] [Feature Requests]
Nowadays NoSQL db are very widespread because they're easy to use. One of the main key advantages is the ability to retrieve a graph/tree instead of doing a lot of JOINs in SQL. But there's an easy solution in SQL: views, and especially materialized views for better performance. Yet, in Supabase, views are not very well supported:

- no realtime support : https://github.com/orgs/supabase/discussions/3824
- no incremental updates extension support: https://github.com/sraoss/pg_ivm
- less supported in Studio: https://github.com/orgs/supabase/discussions/14981

If the solution you recommend is not views but rather a dedicated flat table "One Big Table" or any other solution, it should be said somewhere in the docs.

**Comment by felipe-linares** [comment]
+1 our team is looking to use pg_ivm and didn't see it supported... it's even mentioned in a supabase article despite: https://supabase.com/blog/seen-by-in-postgresql

**Comment by beaupd** [comment]
+1 pg_ivm would also be a solution for a performance problem in my product, would love to see it supported!

**Comment by lookevink** [comment]
bumping this. would love this for performance gains

**Comment by henriquegdantas** [comment]
Bumping this up! Would be awesome for performance gains

**Comment by jaltin** [comment]
Also would love to be able to use pg_ivm.

**Comment by redgriftjosh** [comment]
bumping this +1! I'm using supabase with PowerSync and they recommend pg_ivm for sync_rules with multiple joined tables.

**Reply by jaltin** [reply]
Yes, this is exactly my use case too.

**Reply by redgriftjosh** [reply]
Powersync also says you can use sync_rules with RLS so this is what I'll be trying to use in the meantime.

**Comment by viktorius007** [comment]
bumping this +1! I'm ALSO using supabase with PowerSync and they recommend pg_ivm for sync_rules with multiple joined tables.

**Comment by blakelee** [comment]
Who do I gotta pay to get this integrated? Is there something I or others can do to make this happen? pg_ivm + powersync completely solve a ton of problems that tons of app developers will have with keeping things synced. I can already do this with powersync but pg_ivm gets rid of the denormalization constraints that powersync has.

**Comment by jeff-cap** [comment]
+1 we would use pg_ivm.

**Comment by benarmstrongqualitate** [comment]
+1 for pg_ivm support!

**Comment by rylincoln** [comment]
+1 please
